python/doc/examples/numerical_methods/optimization/plot_ego.py

```diff
@@ -274,7 +275,7 @@
 
 # %%
 # We configure the algorithm :
-algo = ot.EfficientGlobalOptimization(problem, kriging.getResult())
+algo = ot.EfficientGlobalOptimization(problem, kriging.getResult(), noiseModel)
 # assume constant noise var
 guessedNoiseFunction = 0.1
 noiseModel = ot.SymbolicFunction(['x1', 'x2'], [str(guessedNoiseFunction)])
```

Comment:
I can't reach below line 281 with GitHub comment, but I reproduce line 282 here:

```python
algo.setNoiseModel(noiseModel)
```

This is clumsy: first we define a `noiseModel` variable which is actually going to populate the `metamodelNoise_` attribute,  then we redefine `noiseModel` to be a simpler Function before using it to populate the actual `noiseModel_` attribute.

I think the example would be clearer if the complicated noise model were called `metamodelNoise`, and then the EGO were built the following way:

```python
algo = ot.EfficientGlobalOptimization(problem, kriging.getResult())

# %%
# The Kriging metamodel will use the actual noise model.

algo.setMetamodelNoise(metamodelNoise)

# %%
# The criterion however will be computed with a constant noise model, which is much cheaper.

noiseModel = ot.SymbolicFunction(['x1', 'x2'], ['0.1'])
algo.setNoiseModel(noiseModel)
```